Adjust motivating nullness questions (#664)
jspecify/jspecify: docs/docs/spec.md

draw some distinctions, creating at least four kinds of "Is it nullable?"
questions we can ask for any given type usage:

> 1.  What is the [augmented type] of that type usage?

Author: ```suggestion
1. What is the [augmented type] of that type usage?
```

> 2. Do I have to handle the case where null comes out of it?
> 3. Do I have to prevent null from going into it?

Author: ```suggestion
2. Do I have to handle the case where `null` comes out of it?
3. Do I have to prevent `null` from going into it?
```

> 4. Is this type a subtype of that type with respect to nullness?

Reviewer: I could see an argument that this one doesn't fit as neatly into a list of "Is it nullable?" questions. But I think it's close enough: It's kind of about whether one type is "more nullable" than the other. So I think it works.

Reviewer 2: You could phrase it as "Is this type no more nullable than that type, so that it can be a subtype?"

draw some distinctions, creating at least four kinds of "Is it nullable?"
questions we can ask for any given type usage:

> 1.  What is the [nullness operator] of that type usage?
> 2.  Is it reasonable to assume that `null` will not come "out" of it?
> 3.  Is it reasonable to assume that `null` cannot be put "in" to it?

Author: ```suggestion
1.  What is the [augmented type] of that type usage?
2. Do I have to handle the case where null comes out of it?
3. Do I have to prevent null from going into it?
4. Is this type a subtype of that type with respect to nullness?
```

draw some distinctions, creating at least four kinds of "Is it nullable?"
questions we can ask for any given type usage:

1.  What is the [nullness operator] of that type usage?
2.  Is it reasonable to assume that `null` won't come "out" of it?

> 3.  Is it reasonable to assume that `null` can't be put "in" to it?

Reviewer 2: You're going to have to change a word on line 29. 😆
